### PR TITLE
[OS-4]: Create product backend

### PR DESCRIPTION
**Pull request by uladzislau-zhurauski**

### CODE REVIEW

File: shop/dal/product.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    @classmethod
+    def get_all_or_category_products(cls, category=None):
+        return Product.objects.all() if not category else Product.objects.filter(category=category)
+
+    @classmethod
+    def get_product_by_pk(cls, product_pk):
+        try:
+            return Product.objects.get(pk=product_pk)
+        except Product.DoesNotExist:
+            raise Http404
```
Comment (veranika-zakharchenia) [inline shop/dal/product.py:17]: this should be moved to controllers

File: shop/dal/product.py [hunk 2]
```diff
@@ -5,12 +5,13 @@
 
 class ProductDAL:
     @classmethod
-    def get_all_or_category_products(cls, category=None):
-        return Product.objects.all() if not category else Product.objects.filter(category=category)
+    def get_all_or_category_products(cls, category=None, is_available=True):
+        return Product.objects.filter(is_available=is_available) if not category \
```
Comment (veranika-zakharchenia) [inline shop/dal/product.py:9]: rewrite expression without a ternary operator to increase readability

File: shop/controllers/product.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+class ProductController:
+    @classmethod
+    def get_product_list(cls, category_pk):
+        products = ProductDAL.get_all_or_category_products(category_pk)
+        return products
+
+    @classmethod
+    def get_product(cls, product_pk):
+        product = ProductDAL.get_product_by_pk(product_pk)
```
Comment (veranika-zakharchenia) [inline shop/controllers/product.py:12]: where is catch the expection?

Comment (uladzislau-zhurauski) [reply shop/controllers/product.py:12]: It is inside of `get_product_by_pk`

Comment (veranika-zakharchenia) [reply shop/controllers/product.py:12]: we need to separate layer that requests data and logic layer.

**Review by veranika-zakharchenia — APPROVED** [review]

**Pull request merged**